server/api/component_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/pd/v4/server"
+)
+
+var _ = Suite(&testComponentSuite{})
+
+type testComponentSuite struct {
+	svr       *server.Server
+	cleanup   cleanUpFunc
+	urlPrefix string
+}
+
+func (s *testComponentSuite) SetUpSuite(c *C) {
+	s.svr, s.cleanup = mustNewServer(c)
+	mustWaitLeader(c, []*server.Server{s.svr})
+
+	addr := s.svr.GetAddr()
+	s.urlPrefix = fmt.Sprintf("%s%s/api/v1", addr, apiPrefix)
+}
+
+func (s *testComponentSuite) TearDownSuite(c *C) {
+	s.cleanup()
+}
+
+func (s *testComponentSuite) TestComponent(c *C) {
+	// register not happen
+	addr := fmt.Sprintf("%s/component", s.urlPrefix)
+	output := make(map[string][]string)
+	err := readJSON(addr, &output)
+	c.Assert(err, IsNil)
+	c.Assert(len(output), Equals, 0)
+
+	addr1 := fmt.Sprintf("%s/component/c1", s.urlPrefix)
+	var output1 []string
+	err = readJSON(addr1, &output)
+	c.Assert(strings.Contains(err.Error(), "404"), IsTrue)
+	c.Assert(len(output1), Equals, 0)
+
+	// register 2 c1 and 1 c2
+	reqs := []map[string]string{
+		{"component": "c1", "addr": "127.0.0.1:1"},
+		{"component": "c1", "addr": "127.0.0.1:2"},
+		{"component": "c2", "addr": "127.0.0.1:3"},
```

Comment:
Could you add a test case to test domain address?

```suggestion
		{"component": "c2", "addr": "127.0.0.1:3"},
		{"component": "c3", "addr": "example.com"},
```